client/me/help/main.jsx

```diff
@@ -99,7 +108,7 @@ class Help extends React.PureComponent {
 			<div className="help__support-links">
 				<CompactCard
 					className="help__support-link"
-					href="https://support.wordpress.com/"
+					href={ `https://${ getSupportLocale() }.support.wordpress.com` }
```

Comment:
On contrary to the above, I think it's a good call to remove the trailing slash here 👍